src/rod/builder/primitives.py

```diff
@@ -74,34 +82,46 @@ def __post_init__(self) -> None:
             AssertionError: If the scale is not a 3D vector.
             TypeError: If the mesh_path is not a str or pathlib.Path.
         """
+        # Adjust the shape of the scale.
+        self.scale = self.scale.squeeze()
 
-        mesh_path = (
-            self.mesh_path
-            if isinstance(self.mesh_path, pathlib.Path)
-            else pathlib.Path(self.mesh_path)
-        )
+        if self.scale.shape != (3,):
+            raise RuntimeError(f"Scale must be a 3D vector, got '{self.scale.shape}'")
 
-        if not mesh_path.is_file():
-            raise FileNotFoundError(f"Mesh file not found at {self.mesh_path}")
+        # Resolve the mesh URI.
+        mesh_path = resolve_robotics_uri_py.resolve_robotics_uri(uri=str(self.mesh_uri))
 
-        self.mesh: trimesh.base.Trimesh = trimesh.load_mesh(
-            file_obj=mesh_path,
-        )
+        # Build the trimesh object from the mesh path.
+        self.mesh: trimesh.base.Trimesh = trimesh.load_mesh(file_obj=mesh_path)
+
+        # Populate the mass from the mesh if it was not provided externally.
+        if self.mass is None:
+
+            if self.mesh.is_watertight:
+                self.mass = self.mesh.mass
+
+            else:
+                msg = "Mesh '{}' is not watertight. Using a dummy mass value."
+                logging.warning(msg.format(self.mesh_uri))
+                self.mass = 1.0
 
-        assert self.scale.shape == (
-            3,
-        ), f"Scale must be a 3D vector, got {self.scale.shape}"
+        # Populate the inertia tensor from the mesh if it was not provided externally.
+        if self.inertia_tensor is None:
+
+            if self.mesh.is_watertight:
+                self.inertia_tensor = self.mesh.moment_inertia
+
+            else:
+                msg = "Mesh '{}' is not watertight. Using a dummy inertia tensor."
+                logging.warning(msg.format(self.mesh_uri))
+                self.inertia_tensor = np.eye(3)
```

Comment:
Perhaps it is better to just raise an error, instead of specifying a wrong value? So the user can specify the correct inertia if they want, instead of using a wrong value if the warning are overlooked (as it commonly happens)?

Reply:
This was done in this way for speeding up the generation of programmatic models having also meshes when correct inertial data is not needed. Setting the density of the mesh with trimesh is not a big effort, and since we have the support of automatically gathering the inertial properties from the mesh, I'm not against converting this warning to an exception.